TemperatureMeter.c

```diff
@@ -0,0 +1,129 @@
+}*/
+
+static int get_cpu_temp(double *temp) {
+    int n = 0;
+
+    for (const sensors_chip_name *chip = sensors_get_detected_chips(NULL, &n); chip; chip = sensors_get_detected_chips(NULL, &n)) {
+        int r;
+        char buffer[256];
+        r = sensors_snprintf_chip_name(buffer, sizeof(buffer), chip);
+        if (r < 0)
+            continue;
+
+        if (strncmp(buffer, "coretemp", strlen("coretemp")) != 0)
+            continue;
+
+        int m = 0;
+        for (const sensors_feature *feature = sensors_get_features(chip, &m); feature; feature = sensors_get_features(chip, &m)) {
+            if (feature->type != SENSORS_FEATURE_TEMP)
+                continue;
+
+            if (feature->number != 0)
+                continue;
+
+            const sensors_subfeature *sub_feature = sensors_get_subfeature(chip, feature, SENSORS_SUBFEATURE_TEMP_INPUT);
+            if (sub_feature) {
+                double local_temp;
+                r = sensors_get_value(chip, sub_feature->number, &local_temp);
+                if (r == 0) {
+                    *temp = local_temp;
+                    return 0;
+                }
+            }
+        }
+    }
+
+    return -ENOENT;
+}
+
+int TemperatureMeter_attributes[] = {
+   TEMPERATURE_COOL,
+   TEMPERATURE_MEDIUM,
+   TEMPERATURE_HOT,
+};
+
+static void TemperatureMeter_setValues(Meter* this, char* buffer, int len) {
+   ProcessList* pl = this->pl;
+   this->total = pl->totalTasks;
+   this->values[0] = pl->runningTasks;
+   snprintf(buffer, len, "%d/%d", (int) this->values[0], (int) this->total);
+}
+
+static void TemperatureMeter_display(Object* cast, RichString* out) {
+   (void) cast;
+   int r, temperature;
+   double temp;
+
+   r = sensors_init(NULL);
+   if (r < 0) {
+       RichString_append(out, CRT_colors[METER_VALUE], "n/a");
+       return;
+   }
+
+   r = get_cpu_temp(&temp);
+   if (r < 0) {
+       RichString_append(out, CRT_colors[METER_VALUE], "n/a");
+       sensors_cleanup();
+       return;
+   }
+
+   sensors_cleanup();
+
+   int textColor   = CRT_colors[METER_TEXT];
+   int coolColor   = CRT_colors[TEMPERATURE_COOL];
+   int mediumColor = CRT_colors[TEMPERATURE_MEDIUM];
+   int hotColor    = CRT_colors[TEMPERATURE_HOT];
+
+   // convert double to integer
+   temperature = (int)temp;
+
+   // choose the color for the temperature
+   int tempColor;
+   if      (temperature < 60) tempColor = coolColor;
+   else if (temperature < 70) tempColor = mediumColor;
+   else                       tempColor = hotColor;
+
+   // output the temperature
+   char buffer[30];
+   sprintf(buffer, "%d", temperature);
+   RichString_append(out, tempColor, buffer);
+   RichString_append(out, textColor, "°C ");
+}
+
+MeterClass TemperatureMeter_class = {
+   .super = {
+      .extends = Class(Meter),
+      .display = TemperatureMeter_display,
+      .delete = Meter_delete,
+   },
+   .updateValues = TemperatureMeter_setValues,
+   .defaultMode = TEXT_METERMODE,
+   .total = 100.0,
```

Comment:
`100` should be generally a fine max value, however my chip says `crit = +105.0°C`.
We can also get a second temperature in `get_cpu_temp` from `SENSORS_SUBFEATURE_TEMP_MAX` and override the default value set here in `TemperatureMeter_display`
Also when using Fahrenheit this needs to be adjusted.